app/src/main/java/com/sun/ui/loadstate/LoadStateAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.sun.ui.loadstate
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.paging.LoadState
+import androidx.paging.LoadStateAdapter
+import com.sun.BR
+import com.sun.R
+import com.sun.ui.base.BaseViewHolder
+import com.sun.utils.Constants
+
+class LoadStateAdapter(
+    private val listener: LoadStateListener
+) : LoadStateAdapter<LoadStateViewHolder<ViewDataBinding>>() {
+
+    override fun onBindViewHolder(
+        holder: LoadStateViewHolder<ViewDataBinding>,
+        loadState: LoadState
+    ) {
+        holder.binding.apply {
+            Constants.mappingLoadStatus(loadState)
+            when (loadState) {
+                is LoadState.Loading -> {
+                    setVariable(BR.loadState, Constants.LoadStates.LOADING)
+                }
+                is LoadState.Error -> {
+                    setVariable(BR.loadState, Constants.LoadStates.ERROR)
+                }
+                else -> {
+                    setVariable(BR.loadState, Constants.LoadStates.NOT_LOADING)
+                }
+            }
```

Comment:
Đoạn này viết như này thì có đúng logic không nhỉ ? 
```suggestion
        setVariable(BR.loadState, Constants.mappingLoadStatus(loadState))
```